tests/test.py

```diff
@@ -64,10 +67,21 @@ def test():
     # Finally, check if the data is valid JSON
     data = json.loads(data_json)
     assert data["nonce"] == TEST_NONCE, data["nonce"]
-    assert data["url"] == TEST_URL, data["url"]
-    assert len(data["certs"]) > 0, data["certs"]
-    assert len(data["body"]) > 0, data["body"]
+    assert data["url"] == url, data["url"]
+    if expected_status is None:
+        assert data.get("result") == None, data.get("result")
+        assert data["error"]["message"] == "Couldn't connect to server", data["error"]["message"]
+    else:
+        assert data["result"]["status"] == expected_status, data["result"]["status"]
+        assert len(data["result"]["certs"]) > 0, data["result"]["certs"]
+        assert len(data["result"]["body"]) > 0, data["result"]["body"]
+        assert data.get("error") == None, data.get("error")
```

Comment:
I think `"error" not in data` is a bit clearer than `get("error") == None` (ie. what if `error` was present but null? is that allowed behaviour?)